src/backends/tensorrt/tensorrtlib.cc

```diff
@@ -196,6 +199,21 @@ namespace dd
             + this->_mlmodel._repo);
       }
 
+    if (ad.has("template"))
+      {
+        std::string tmplate = ad.get("template").get<std::string>();
+        this->_logger->info("Model template is {}", tmplate);
+
+        if (tmplate == "yolox")
+          {
+            this->_mltype = "detection";
+            _need_nms = true;
+            _top_k = 200; // XXX: pass as parameter?
```

Comment:
This should be part of the model yes, or of an external config struct for yolox models, outside of tensorrtlib ?

Reply:
With caffe model we can read this information directly in the model. With onnx export I think it is way harder. So we would need a model config file somewhere?

Reply:
Let's put it into the AP instead (output object with a default value of 200 or 400?)

Reply:
it's in `parameters.mllib.topk` and must be defined during `init` (because it's dependent on the model, and it can't change between predict calls)